app/imports/collections/applications.js

```diff
@@ -0,0 +1,19 @@
+import { Meteor } from 'meteor/meteor';
+
+class ApplicationsCollection extends Mongo.Collection {
+  insert(doc, callback) {
+    doc.createdAt = doc.createdAt || new Date();
+    doc.updatedAt = doc.updatedAt || new Date();
+    doc.userId = Meteor.userId();
+    doc.year = doc.year || new Date().getFullYear();
+
+    return super.insert(doc,callback);
```

Comment:
no

`this` refers to itself (ApplicationsCollection) while `super` refers to the parent extended class (Mongo.Collection)